src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -64,12 +64,14 @@ public abstract class NodeModel : ModelBase, IRenderPackageSource<NodeModel>, ID
         private ObservableCollection<PortModel> inPorts = new ObservableCollection<PortModel>();
         private ObservableCollection<PortModel> outPorts = new ObservableCollection<PortModel>();
 
+        private readonly Dictionary<int, Tuple<int, NodeModel>> inputNodes;
+        private readonly Dictionary<int, HashSet<Tuple<int, NodeModel>>> outputNodes;
+
         #endregion
 
-        #region public members
+        internal const double NodeHeaderHeight = 25;
```

Comment:
Given the X, Y are already obsoleted, it seems strange we keep adding this on NodeModel level, is there other options?

Reply:
This is also needed in `portmodel` so if I add it to `NodeViewModel`, it won't be available to the `portmodel`.

Reply:
Also, I don't think we can obsolete X, Y. From my understanding, they are still necessary and even the `NodeViewModel`, X, and Y properties eventually derive from those of the `NodeModel`.